substrafl/algorithms/pytorch/weight_manager.py

```diff
@@ -93,7 +93,7 @@ def get_parameters(
     Returns:
         typing.List[torch.nn.parameter.Parameter]: The list of torch parameters of the provided model.
     """
-    with torch.inference_mode():
+    with torch.no_grad():
```

Comment:
From a cursory search, it seems that `torch.inference_mode` is the [newest, recommended way](https://pytorch.org/docs/1.9.0/notes/autograd.html#inference-mode) of performing inferences, even if it indeed does not allows for further training on the model. 
It is supposed to bring some performance improvements, it might be minimal but it's probably worth investigating.

Reply:
Fair enough